src/boot_O2_g3/fault.c

```diff
@@ -182,98 +239,111 @@ void Fault_RemoveAddrConvClient(FaultAddrConvClient* client) {
+ * L toggles auto-scroll
+ * DPad-Up enables osSyncPrintf output
+ * DPad-Down disables osSyncPrintf output
+ * A and DPad-Right continues and returns true
+ * DPad-Left continues and returns false
+ */
+u32 Fault_WaitForInputImpl(void) {
+    Input* input = &sFaultInstance->inputs[0];
     s32 count = 600;
-    u32 kDown;
+    u32 pressedBtn;
 
-    while (1) {
-        Fault_Sleep(0x10);
+    while (true) {
+        Fault_Sleep(1000 / 60);
         Fault_UpdatePadImpl();
 
-        kDown = curInput->press.button;
+        pressedBtn = input->press.button;
 
-        if (kDown == BTN_L) {
-            sFaultContext->faultActive = !sFaultContext->faultActive;
+        if (pressedBtn == BTN_L) {
+            sFaultInstance->autoScroll = !sFaultInstance->autoScroll;
         }
 
-        if (sFaultContext->faultActive) {
+        if (sFaultInstance->autoScroll) {
             if (count-- < 1) {
-                return 0;
+                return false;
             }
         } else {
-            if (kDown == BTN_A || kDown == BTN_DRIGHT) {
-                return 0;
+            if (pressedBtn == BTN_A || pressedBtn == BTN_DRIGHT) {
+                return false;
             }
 
-            if (kDown == BTN_DLEFT) {
-                return 1;
+            if (pressedBtn == BTN_DLEFT) {
+                return true;
             }
 
-            if (kDown == BTN_DUP) {
-                FaultDrawer_SetOsSyncPrintfEnabled(1);
+            if (pressedBtn == BTN_DUP) {
+                FaultDrawer_SetOsSyncPrintfEnabled(true);
             }
 
-            if (kDown == BTN_DDOWN) {
-                FaultDrawer_SetOsSyncPrintfEnabled(0);
+            if (pressedBtn == BTN_DDOWN) {
+                FaultDrawer_SetOsSyncPrintfEnabled(false);
             }
         }
     }
 }
 
-void Fault_WaitForInput() {
+void Fault_WaitForInput(void) {
     Fault_WaitForInputImpl();
 }
 
 void Fault_DrawRec(s32 x, s32 y, s32 w, s32 h, u16 color) {
     FaultDrawer_DrawRecImpl(x, y, x + w - 1, y + h - 1, color);
 }
 
-void Fault_FillScreenBlack() {
-    FaultDrawer_SetForeColor(0xFFFF);
-    FaultDrawer_SetBackColor(1);
+void Fault_FillScreenBlack(void) {
+    FaultDrawer_SetForeColor(GPACK_RGBA5551(255, 255, 255, 1));
+    FaultDrawer_SetBackColor(GPACK_RGBA5551(0, 0, 0, 1));
     FaultDrawer_FillScreen();
-    FaultDrawer_SetBackColor(0);
+    FaultDrawer_SetBackColor(GPACK_RGBA5551(0, 0, 0, 0));
 }
 
-void Fault_FillScreenRed() {
-    FaultDrawer_SetForeColor(0xFFFF);
-    FaultDrawer_SetBackColor(0xF001);
+void Fault_FillScreenRed(void) {
+    FaultDrawer_SetForeColor(GPACK_RGBA5551(255, 255, 255, 1));
+    FaultDrawer_SetBackColor(GPACK_RGBA5551(240, 0, 0, 1));
     FaultDrawer_FillScreen();
-    FaultDrawer_SetBackColor(0);
+    FaultDrawer_SetBackColor(GPACK_RGBA5551(0, 0, 0, 0));
 }
 
 void Fault_DrawCornerRec(u16 color) {
-    Fault_DrawRec(0x16, 0x10, 8, 1, color);
+    Fault_DrawRec(22, 16, 8, 1, color);
 }
 
 void Fault_PrintFReg(s32 idx, f32* value) {
```

Comment:
```suggestion
void Fault_PrintFReg(s32 index, f32* value) {
```
I think we've been trying to purge `idx` -> `index`?